arduino/test/health_monitor_core_test.cpp

```diff
@@ -0,0 +1,263 @@
+
+    static constexpr int kExpectedIterToZombie = 59;
+    for (int i = 0; i < kExpectedIterToZombie; ++i) {
+        auto const next_health = exposure_update(curr_health,
+                                                 ExposureEvent{
+                                                     .human = 3,
+                                                     .cat   = 0,
+                                                 });
+        ASSERT_GT(next_health.health, curr_health.health);
+        curr_health = next_health;
+        if (is_zombie(next_health.health)) {
+            break;
+        }
+    }
+
+    ASSERT_GT(curr_health.health, to_health(StateBounds::INFECTED_ASYM));
+    EXPECT_GE(curr_health.health, to_health(StateBounds::ZOMBIE));
+}
+
+TEST(ExposureUpdateTests, CatExposureCausesCatResistance) {
+    HealthState curr_health{.health = to_health(StateBounds::HEALTHY), .cat_resistance = false};
+
+    static constexpr int kMaxIterToCatResist = 10;
+    for (int i = 0; i < kMaxIterToCatResist; ++i) {
+        auto const next_health = exposure_update(curr_health,
+                                                 ExposureEvent{
+                                                     .human = 0,
+                                                     .cat   = 1,
+                                                 });
+        ASSERT_GT(next_health.health, curr_health.health);
+        bool const cat_resistance_changed =
+            next_health.cat_resistance != curr_health.cat_resistance;
+        curr_health = next_health;
+        if (cat_resistance_changed) {
+            break;
+        }
+    }
+
+    ASSERT_TRUE(curr_health.cat_resistance);
+    ASSERT_GT(curr_health.health, to_health(StateBounds::HEALTHY));
+    EXPECT_GE(curr_health.health, to_health(StateBounds::INFECTED_ASYM));
+}
+
+TEST(HealthMonitorCoreTests, NewPlayer) {
+    const PlayerState player = new_player_state();
+    EXPECT_EQ(player.tick, 0);
+    EXPECT_EQ(player.health.health, to_health(StateBounds::SUPER_HEALTHY));
+    EXPECT_FALSE(player.health.cat_resistance);
+}
+
+TEST(HealthMonitorCoreTests, SingleExposureEvent) {
+    const PlayerState player_init = new_player_state();
+    PlayerState player            = player_init;
+
+    int iterations              = 0;
+    int exposure_events_handled = 0;
+    int treament_events_handled = 0;
+    game_update(
+        player,
+        [&iterations]() -> Event {
+            const auto it = iterations++;
+            if (it < 1) {
+                return Event{ExposureEvent{.human = 3, .cat = 1}};
+            } else {
+                return Event{};
+            }
+        },
+        [&exposure_events_handled](ExposureEvent const& exposure_event) {
+            ++exposure_events_handled;
+        },
+        [&treament_events_handled](TreatmentEvent const& exposure_event) {
+            ++treament_events_handled;
+        });
+
+    ASSERT_EQ(exposure_events_handled, 1);
```

Comment:
Interesting. I see that in the code this is what makes sense but I would have assumed that this number should have been 3

Reply:
maybe I'm too lost in my own sauce, but I can't figure out why 3